pkg/external-provider/provider.go

```diff
@@ -54,7 +55,14 @@ func (p *externalPrometheusProvider) GetExternalMetric(ctx context.Context, name
 	queryResults, err := p.promClient.Query(ctx, pmodel.Now(), selector)
 
 	if err != nil {
-		klog.Errorf("unable to fetch metrics from prometheus: %v", err)
+		if promErr, ok := err.(*prom.Error); ok { // Check if the error is of type *prom.Error
```

Comment:
ditto: Shouldn't we also be checking if this is of type `*client.Error` instead?

Reply:
**Actually, I think we should make `promClient.Query` always return a `*client.Error` instead**